File: src/algorithms/ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, int> dict;
        for (int i = 0; i < nums.size(); i++) {
            int t = nums[i];
            if (dict[t] > 0) return true;
            dict[t] = 1;
        }
        return false;
    }
};

File: Algorithms/python/217-ContainsDuplicate/ContainsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums_count = dict()

        for v in nums:
            if v not in nums_count:
                nums_count[v] = 1
                continue
            c = nums_count[v]
            if c > 0:
                return True
            nums_count[v] = c + 1

        return False
